File: sofiia_tuboltseva/bridge/Bridge2/shape/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge2
{
    // Abstraction
    abstract class Shape
    {
        protected IColor color;//захищене

        public Shape(IColor color)
        {
            this.color = color;//this посилання
        }

        public abstract void Draw();
        public abstract double GetArea();
        public abstract double GetPerimeter();
    }
}

File: bryniarskyi_nazarii/bridge/shape/Shape.java
package org.example.bridge.shape;

import org.example.bridge.color.Color;

public abstract class Shape {

    private final Color color;

    public Shape(Color color) {
        this.color = color;
    }

    public abstract double getSquare();

    public abstract double getPerimeter();

    public void drawShape() {
        color.draw();
    }
}
